src/main/java/games/strategy/triplea/ai/proAI/data/ProTerritoryManager.java

```diff
@@ -188,21 +187,21 @@ private static void findScrambleOptions(final PlayerID player, final Map<Territo
     }
     final Match<Unit> airbasesCanScramble = Match.all(Matches.unitIsEnemyOf(data, player),
         Matches.UnitIsAirBase, Matches.UnitIsNotDisabled, Matches.unitIsBeingTransported().invert());
-    final CompositeMatchAnd<Territory> canScramble = new CompositeMatchAnd<>(
-        Match.any(Matches.TerritoryIsWater, Matches.isTerritoryEnemy(player, data)),
-        Matches.territoryHasUnitsThatMatch(Match.all(Matches.UnitCanScramble,
-            Matches.unitIsEnemyOf(data, player), Matches.UnitIsNotDisabled)),
-        Matches.territoryHasUnitsThatMatch(airbasesCanScramble));
+    final Match.CompositeBuilder<Territory> canScrambleBuilder = Match.<Territory>newCompositeBuilder()
+        .add(Match.any(Matches.TerritoryIsWater, Matches.isTerritoryEnemy(player, data)))
+        .add(Matches.territoryHasUnitsThatMatch(Match.all(Matches.UnitCanScramble,
+            Matches.unitIsEnemyOf(data, player), Matches.UnitIsNotDisabled)))
+        .add(Matches.territoryHasUnitsThatMatch(airbasesCanScramble));
```

Comment:
I think for these complex ones, its better to go ahead and do some additional formatting if you don't mind. As otherwise they are very difficult to read.